Model/Api/DiscountCodeValidation.php

```diff
@@ -708,8 +758,8 @@ private function convertToBoltDiscountType($type)
     private function setCouponCode($quote, $couponCode)
     {
         $quote->getShippingAddress()->setCollectShippingRates(true);
-        $quote->setCouponCode($couponCode);
-        $this->cartHelper->saveQuote($quote->collectTotals());
+        $quote->setCouponCode($couponCode)->collectTotals()->save();
+        $this->cartHelper->saveQuote($quote);
```

Comment:
I think we should not save twice the same object!

Reply:
What is the reason to add here
`$quote->setCouponCode($couponCode)->collectTotals()->save();`
additional save ?